### PR TITLE
Load "x-ms-skip-url-encoding" from extension and fix type property skip

### PR DESCRIPTION
**Pull request by live1206**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by ArthurMa1978 — APPROVED** [review]

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts [hunk 1]
```diff
@@ -306,7 +307,10 @@ export function loadOperation(
             IsResourceParameter: false,
             IsContentType: isContentType,
             IsEndpoint: false,
-            SkipUrlEncoding: false, //TODO: retrieve out value from extension
+            SkipUrlEncoding:
+                // TODO: update this when https://github.com/Azure/typespec-azure/issues/1022 is resolved
```
Comment (live1206) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts:311]: Update after https://github.com/Azure/typespec-azure/issues/1022 resolved

File: src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs [hunk 2]
```diff
@@ -474,6 +474,13 @@ protected override IEnumerable<ObjectTypeProperty> BuildProperties()
                         continue;
                     }
                     var prop = CreateProperty(property);
+
+                    // If "Type" property is "String" and "ResourceType" exists in parents properties, skip it since they are the same.
+                    // This only applies for TypeSpec input, for swagger input we have renamed "type" property to "resourceType" in FrameworkTypeUpdater.ValidateAndUpdate
+                    if (property.CSharpName() == "Type" && prop.ValueType.Name == "String" && existingProperties.Contains("ResourceType"))
+                    {
+                        continue;
+                    }
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs:483]: OK this is really like a workaround.
adding a client.tsp in the resourcemanager to change the client to ResourceType in all trackedresource and resource may be a better idea

Comment (live1206) [reply src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs:483]: Ideally, we should rename this for csharp for common-types, but for now we don't have an entry for common-types.
We will need to have an entry to be able to generate `Azure.ResourceManager` from common-types typespec.

**Comment by live1206** [comment]
> Looks we only implements this workaround in dotnet for 'x-ms-skip-url-encoding', it would cause inconsistency around SDKs for the behavior of this extension. Considering there is no service impact by this now, shall we rollback this change and wait for the decision for [Azure/typespec-azure#1022](https://github.com/Azure/typespec-azure/issues/1022)? cc @m-nash @live1206

`Scvmm` is affected by this, .net generator failed without this fix, not like other languages could generate SDK with this part missing.
This time, we will release GA version for `Scvmm` so it generates from swagger, but I would prefer to keep the fix until we have a proper solution. And, we will talk about this in the coming meeting tomorrow.

**Pull request merged**